src/main/java/uk/gov/hmcts/divorce/document/print/LetterPrinter.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.divorce.document.print;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.divorce.divorcecase.model.Applicant;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.document.DocumentGenerator;
+import uk.gov.hmcts.divorce.document.print.documentpack.DocumentPackInfo;
+import uk.gov.hmcts.divorce.document.print.model.Letter;
+import uk.gov.hmcts.divorce.document.print.model.Print;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class LetterPrinter {
+
+    private final DocumentGenerator documentGenerator;
+    private final BulkPrintService bulkPrintService;
+
+    public void sendLetters(final CaseData caseData,
```

Comment:
Should it return CaseData to indicate it's mutated it?

Reply:
I would say yes but since it would just get returned to the ApplicantNotification interface which is void, it's a bit pointless without doing a full refactor of every implementer (in essence, all notification dispatches)